go/exorelay/exorelay.go

```diff
@@ -65,27 +65,30 @@ func (e *ExoRelay) GetMessageChannel() chan structs.Message {
 }
 
 // Send sends a message with the given options
-func (e *ExoRelay) Send(options MessageOptions) (string, error) {
+func (e *ExoRelay) Send(options MessageOptions) (*structs.Message, error) {
 	if e.socket == nil {
-		return "", errors.New("ExoRelay#Send not connected to Exocom")
+		return &structs.Message{}, errors.New("ExoRelay#Send not connected to Exocom")
 	}
 	id := uuid.NewV4().String()
 	if options.Name == "" {
-		return "", errors.New("ExoRelay#Send cannot send empty messages")
+		return &structs.Message{}, errors.New("ExoRelay#Send cannot send empty messages")
+	}
+	if options.ActivityID == "" {
+		options.ActivityID = uuid.NewV4().String()
 	}
```

Comment:
Thoughts on creating a local activityID variable instead of updating the passed in options? I prefer to not mutate inputs. Not a big deal in the current case as we are getting a copy of the messageOptions but if we ever changed it to a pointer then we would be updating the passed in options. This could be bad if the caller wanted to use it again after and wasn't expecting this function to change it.